code/ui/blocks/src/blocks/Canvas.tsx

```diff
@@ -54,13 +124,88 @@ const usePreviewProps = (
   };
 };
 
-export const Canvas: FC<CanvasProps> = (props) => {
+export const Canvas: FC<CanvasProps & DeprecatedCanvasProps> = (props) => {
   const docsContext = useContext(DocsContext);
   const sourceContext = useContext(SourceContext);
-  const { isLoading, previewProps } = usePreviewProps(props, docsContext, sourceContext);
-  const { children } = props;
+  const { children, of, source } = props;
+  const { isLoading, previewProps } = useDeprecatedPreviewProps(props, docsContext, sourceContext);
+  let story;
+  let sourceProps;
+  try {
+    /**
+     * useOf and useSourceProps will throw if they can't find the story, in the scenario where
+     * the doc is unattached (no primary story) and 'of' is undefined.
+     * That scenario is valid in the deprecated API, where children is used as story refs rather than 'of'.
+     * So if children is passed we allow the error to be swallowed and we'll use them instead.
+     */
+    ({ story } = useOf(of || 'story', ['story']));
+    sourceProps = useSourceProps({ ...source, of }, docsContext, sourceContext);
+  } catch (error) {
+    if (!children) {
+      throw error;
+    }
+  }
```

Comment:
It's [debatable](https://github.com/facebook/react/issues/16026) whether or not this follows the rules of hooks, as technically the try-block could work as a condition, that could cause the number of hooks called to change between renders - if `useOf` throws in one render but not in another, then React will fail.

I'm unsure how to support the described scenario in a cleaner way that this.

Reply:
that would at least break the rule "less", but purists would still not consider it safe.
but you're right, I'll do that.